server/routerlicious/packages/services-shared/src/livenessMonitor/livenessMonitor.ts

```diff
@@ -0,0 +1,60 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { IChecker, ILivenessMonitor } from "@fluidframework/server-services-core";
+import { LumberEventName, Lumberjack } from "@fluidframework/server-services-telemetry";
+import { Router } from "express";
+import { handleResponse } from "../http";
+
+export class LivenessMonitor implements ILivenessMonitor {
+	private static instance: LivenessMonitor;
+	private readonly checkerMap: Map<string, () => Promise<void>> = new Map();
+
+	public static getInstance(checkList: IChecker[]): LivenessMonitor {
```

Comment:
>getInstance(checkList: IChecker[] [](http://example.com/codeflow?start=15&length=33)

Not sure if this is a common pattern. The API could be misleading since checkList is only used for the first call and is ignored for the subsequent calls.

Reply:
Since we have registerChecker(), an alternative design is not passing checkList to getInstance